service/frontend/handler.go

```diff
@@ -273,15 +300,69 @@ func (wh *WorkflowHandler) UpdateDomain(ctx context.Context,
 	info := getResponse.Info
 	config := getResponse.Config
 	replicationConfig := getResponse.ReplicationConfig
-	clusterMetadate := wh.GetClusterMetadata()
-	currentCluster := clusterMetadate.GetCurrentClusterName()
-	isCurrentClusterActive := currentCluster == replicationConfig.ActiveClusterName
+	configVersion := getResponse.ConfigVersion
 	failoverVersion := getResponse.FailoverVersion
+	currentCluster := clusterMetadata.GetCurrentClusterName()
 
 	// whether active cluster is changed
 	activeClusterChanged := false
 	// whether anything other than active cluster is changed
 	configurationChanged := false
+
+	validateReplicationConfig := func(existingDomain *persistence.GetDomainResponse,
```

Comment:
nit: Why this validation is defined as a closure? Can we just have a validation function defined on the handler like we do for all different kind of validation?

Reply:
the reason i put this validation as a closure: this function is not reusable, by other logics.

but sure, i will make the change